src/main/java/duke/command/DoneCommand.java

```diff
@@ -0,0 +1,52 @@
+package duke.command;
+
+import java.io.IOException;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.exception.DukeInvalidIndexException;
+
+/**
+ * Represents a marking a task as done action.
+ */
+public class DoneCommand implements Command {
+    private final int index;
+
+    /**
+     * Class constructor with specified index.
+     *
+     * @param index The index of the Task to be marked as done.
+     */
+    public DoneCommand(int index) {
+        this.index = index;
+    }
+
+    /**
+     * Returns false because command does not exit.
+     *
+     * @return false.
+     */
+    @Override
+    public boolean shouldExit() {
```

Comment:
I really like how you keep this shouldExit() as a method and override it in child class!

Reply:
Thank you!